### ISSUE TITLE
Code for Back Propagation in CNNs

### ISSUE THREAD

**Issue by kelefouras** [oneapi-src/oneDNN#1641]
Dear all

I hope this message finds you well

In the provided example for training CNNs the back propagation phase is not included (phase where the gradients are calculated). This is realized with convolution_backward_data(). Instead, in the example provided only the gradient descent is provided which is realized with convolution_backward_weights(). 

Is there any example that includes the back propagation phase?

thank you
Vasilis

**Comment by igorsafo** [comment]
Hello @kelefouras ,
Yes, You are right, the example doesn't include the back propagation phase for the convolution, because this is the first layer in the example. I checked and there is no an example with the back propagation phase for convolution, but you can look into oneDNN gtest for the back propagation phase of convolution: https://github.com/oneapi-src/oneDNN/blob/8e86fa9088e3a423e468a6c36046f8cbf7b967a0/tests/gtests/test_convolution_backward_data_common.hpp#L232-L259

It is similar to the convolution_backward_weights(), the differences are:
- A different class: convolution_backward_data()
- A different set of arguments: convolution_backward_data() requires DNNL_ARG_DIFF_SRC, DNNL_ARG_DIFF_DST, DNNL_ARG_WEIGHTS & DNNL_ARG_BIAS

These operations are independent so they can be called in any order and their arguments do not depend on each other.
